shared/src/main/scala/kinesis4cats/producer/fs2/FS2Producer.scala

```diff
@@ -72,17 +72,19 @@ abstract class FS2Producer[F[_], PutReq, PutRes](implicit
     for {
       _ <- logger.debug(ctx.context)("Received record to put")
       deferred <- Deferred[F, F[Producer.Result[PutRes]]]
-      res <- channel.send(record -> deferred)
-      _ <- res.bitraverse(
-        _ =>
-          logger.warn(ctx.context)(
-            "Producer has been shut down and will not accept further requests"
-          ),
-        _ =>
-          logger.debug(ctx.context)(
-            "Successfully put record into processing queue"
-          )
-      )
+      res <- channel.send(record -> deferred).race(channel.closed)
```

Comment:
According to `Channel` docs, this is required, for a case when someone was hanging on `send`, but the channel got closed (due to cancellation). Without the race here, it will hang forever.